.github/workflows/newman.yml

```
@@ -0,0 +1,147 @@
+on:
+  workflow_call:
+    inputs:
+      collection_path:
+        required: true
+        type: string
+      container_name:
+        required: true
+        type: string
+      centreon_url:
+        required: true
+        type: string
+      centreon_image:
+        required: true
+        type: string
+    secrets:
+      registry_username:
+        required: true
+      registry_password:
+        required: true
+
+jobs:
+  discover_collections:
+    runs-on: ubuntu-latest
+
+    outputs:
+      collections: ${{ steps.set_collections.outputs.collections }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: List Postman Collections and Environments
+        id: set_collections
+        run: |
+          collection_path="${{ inputs.collection_path }}"
+          collections=$(find "$collection_path" -type f -name "*.postman_collection.json" -printf "%P\n" | sort | jq -R -s -c 'split("\n")[:-1]')
+          echo "collections=$collections" >> $GITHUB_OUTPUT
+          echo "GITHUB_OUTPUT contents:"
+          cat $GITHUB_OUTPUT
+
+  postman_tests:
+   needs: discover_collections
+   runs-on: ubuntu-latest
+
+   strategy:
+    fail-fast: false
+    matrix:
+      feature: ${{ fromJson(needs.discover_collections.outputs.collections) }}
+
+   name: Postman Test - ${{ matrix.feature }}
+
+   defaults:
+    run:
+      shell: bash
+      working-directory: centreon/tests/rest_api
+   env:
+    CONTAINER_NAME: ${{ inputs.container_name }}
+    CENTREON_URL: ${{ inputs.centreon_url }}
+    CENTREON_IMAGE: ${{ inputs.centreon_image }}
+
+   steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v4
+
+    - uses: pnpm/action-setup@v2
+      with:
+        version: 7
+
+    - name: Install Dependencies
+      run: pnpm install --frozen-lockfile
+      working-directory: centreon/tests/rest_api

```

this will install dependencies faster
```suggestion
      working-directory: centreon/tests/rest_api
      env:
          CYPRESS_INSTALL_BINARY: "0"
          PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: "1"
          PUPPETEER_SKIP_CHROMIUM_DOWNLOAD: "true"

```